### PR TITLE
Ports TG's drop and pickup animations

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Title.
TG has simple little animations for picking up items and dropping them, this PR ports them.

## Why It's Good For The Game

It looks fancy and lets you see who's picking up what.

## Changelog
:cl:
add: Ported TG's animation effects for dropping and picking up items.
/:cl:

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -652,6 +652,83 @@
 	if(SSmapping.configs[GROUND_MAP].environment_traits[MAP_COLD] && (flags_item_map_variant & ITEM_ICE_PROTECTION))
 		min_cold_protection_temperature = ICE_PLANET_MIN_COLD_PROTECTION_TEMPERATURE
 
+/**
+	Play small animation and jiggle when picking up an object
+ */
+
```
Comment (TiviPlus) [inline code/game/objects/items.dm:658]:
```suggestion
/**
 * Play small animation and jiggle when picking up an object
 */
```

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -652,6 +652,83 @@
...
+/obj/item/proc/do_pickup_animation(atom/target)
+	set waitfor = FALSE
+	if(!istype(loc, /turf))
+		return
```
Comment (TiviPlus) [inline code/game/objects/items.dm:662]: isturf

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -652,6 +652,83 @@
 	if(SSmapping.configs[GROUND_MAP].environment_traits[MAP_COLD] && (flags_item_map_variant & ITEM_ICE_PROTECTION))
 		min_cold_protection_temperature = ICE_PLANET_MIN_COLD_PROTECTION_TEMPERATURE
 
+/**
+	Play small animation and jiggle when picking up an object
+ */
+
+/obj/item/proc/do_pickup_animation(atom/target)
+	set waitfor = FALSE
```
Comment (TiviPlus) [inline code/game/objects/items.dm:660]: note to self

File: code/game/objects/items.dm [hunk 4]
```diff
@@ -652,6 +652,83 @@
...
+	var/matrix/animation_matrix = new
+	animation_matrix.Turn(pick(-30, 30))
+	animate(pickup_animation, alpha = 175, pixel_x = to_x, pixel_y = to_y, time = 3, transform = animation_matrix, easing = CUBIC_EASING)
+	sleep(1)
+	animate(pickup_animation, alpha = 0, transform = matrix(), time = 1)
+
+/**
+	Play small animation and jiggle when dropping an object
+ */
+
```
Comment (TiviPlus) [inline code/game/objects/items.dm:694]:
```suggestion
/**
 * Play small animation and jiggle when dropping an object
 */
```

File: code/game/objects/items.dm [hunk 5]
```diff
@@ -652,6 +652,83 @@
...
+/obj/item/proc/do_drop_animation(atom/moving_from)
+	if(!istype(loc, /turf))
+		return
```
Comment (TiviPlus) [inline code/game/objects/items.dm:697]: isturf

File: code/modules/mob/inventory.dm [hunk 6]
```diff
@@ -43,6 +43,8 @@
 	Returns TRUE on success.
 */
 /mob/proc/put_in_l_hand(obj/item/W)
+	if(isturf(W.loc))
+		W.do_pickup_animation(src)
```
Comment (TiviPlus) [inline code/modules/mob/inventory.dm:47]: why add these isturf checks here and put in r hand if its already checked in the proc?

File: code/modules/mob/inventory.dm [hunk 7]
```diff
@@ -128,6 +132,9 @@
 	Returns TURE if it was able to put the thing into one of our hands.
 */
 /mob/proc/put_in_hands(obj/item/W, del_on_fail = FALSE)
+	log_world("Putting[W] in hands")
```
Comment (TiviPlus) [inline code/modules/mob/inventory.dm:135]:
```suggestion
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm (hunk 3 above)
Comment (RipGrayson) [reply code/game/objects/items.dm:658]: I actually don't know what's being requested here.

Comment (BraveMole) [reply code/game/objects/items.dm:658]: Nothing

File: code/game/objects/items.dm [hunk 8]
```diff
@@ -653,6 +653,79 @@
...
+		to_x += 32
+	else if(direction & WEST)
+		to_x -= 32
+	if(!direction)
+		to_y += 16
+	flick_overlay(pickup_animation, GLOB.clients, 6)
+	var/matrix/animation_matrix = new
+	animation_matrix.Turn(pick(-30, 30))
+	animate(pickup_animation, alpha = 175, pixel_x = to_x, pixel_y = to_y, time = 3, transform = animation_matrix, easing = CUBIC_EASING)
+	sleep(1)
```
Comment (BraveMole) [inline code/game/objects/items.dm:686]: Sleep seems acceptable here, considering it is just an animate behind, right tivi?

**Review by BraveMole — APPROVED** [review]

File: code/game/objects/items.dm (hunk 8 above)
Comment (TiviPlus) [reply code/game/objects/items.dm:686]: No it should be three animate calls I need to fix it on tg

File: code/game/objects/items.dm (hunk 2 above)
Comment (TiviPlus) [reply code/game/objects/items.dm:659]:
```suggestion
	if(!isturf(loc))
		return
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**